### DISCUSSION TITLE
Source Control/Git

### DISCUSSION THREAD

**Discussion by Aging-Developer** [supabase/supabase#20746] [Questions] [answered]
Hi,

Is there a simple way to connect my supabase DB to a source control system(git repo), so all schema (tables, views, functions) can be pushed to the repo so to prevent accident loss and version history?

**Comment by charislam** [comment, marked as answer]
It sounds like what you want is a migration history (basically a set of SQL files that define how your database got its current schema, which if run again in order will reproduce your schema from scratch).

Supabase supports migration files through the CLI, including pulling the current remote schema to create a migration file if you don't have any, and creating new migration files. Take a look at the [migration CLI docs](https://supabase.com/docs/reference/cli/supabase-db-pull), and don't hesitate to follow up if you have questions!
